test/machi_file_proxy_eqc.erl

```diff
@@ -224,28 +316,35 @@ write_args(S) ->
     {Off, Len} = hd(S#state.planned_writes),
     [S#state.pid, Off, data_with_csum(Len)].
 
-write_ok(_S, [_Pid, Off, _Data]) when Off < 1024 -> false;
 write_ok(S, [_Pid, Off, {Bin, _Tag, _Csum}]) ->
     Size = iolist_size(Bin),
     %% Check writes checks if a byte range is *written*
     %% So writes are ok IFF they are NOT written, so
     %% we want not check_writes/3 to be true.
-    check_writes(write, S#state.written, Off, Size).
+    case {get_overlaps(Off, Size, S#state.written, []),
+          get_overlaps(Off, Size, S#state.trimmed, [])} of
+        {[], []} ->
+            true;
+        {[{Off, Size}], []} ->
+            true;
+        _Other ->
+            false
+    end.
 
 write_post(S, Args, Res) ->
     case write_ok(S, Args) of
         %% false means this range has NOT been written before, so
         %% it should succeed
-        false -> eq(Res, ok);
+        true -> eq(Res, ok);
         %% mostly true means we've written this range before BUT
         %% as a special case if we get a call to write the EXACT
         %% same data that's already on the disk, we return "ok"
         %% instead of {error, written}.
-        mostly_true -> probably_error(Res); 
+        %% mostly_true -> probably_error(Res); 
```

Comment:
Should be removed? If it is nice for debugging, it's OK to be left as commented.